pkg/k8s/apis/cilium.io/v1/register.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"time"
+
+	k8sconst "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
+
+	"k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// ThirdPartyResourcesSingularName is the singular name of third party resources
+	ThirdPartyResourcesSingularName = "cilium-network-policy"
+
+	// ThirdPartyResourceGroup is the name of the third party resource group
+	ThirdPartyResourceGroup = k8sconst.GroupName
+
+	// CustomResourceDefinitionVersion is the current version of the resource
+	ThirdPartyResourceVersion = "v1"
+)
+
+// SchemeGroupVersion is group version used to register these objects
+var SchemeGroupVersion = schema.GroupVersion{
+	Group:   ThirdPartyResourceGroup,
+	Version: ThirdPartyResourceVersion,
+}
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+var (
+	SchemeBuilder runtime.SchemeBuilder
```

Comment:
Please add comments to variables needed by the auto generated code

Reply:
Done